### PR TITLE
docs(hydrate): clean up the docs for the hydrate app output target

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
Just some cleanup to clear up confusing wording and explanations here,
format things a little bit more cleanly, and so on.

### CODE REVIEW

File: src/docs/guides/hydrate-app.md [hunk 1]
```diff
@@ -10,23 +10,52 @@ contributors:
 
 # Hydrate App
 
-The hydrate app is a bundle of your same components, but compiled so they can be hydrated on a NodeJS server, and generate HTML. This is the same NodeJS app which prerendering uses internally, and the same one which Angular Universal SSR uses for Ionic. Like Stencil components, the hydrate app itself is not restricted to one framework.
+The hydrate app is a Stencil output target which generates a module that can be
+used on a NodeJS server to hydrate HTML and implement server side rendering (SSR).
+This functionality is used internally by the Stencil compiler for
+prerendering, as well as for the Angular Universal SSR for the Ionic
+framework. However, like Stencil components, the hydrate app itself is not
```
[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply src/docs/guides/hydrate-app.md:17]: (Indeed it is)

File: src/docs/guides/hydrate-app.md [hunk 2]
```diff
@@ -10,23 +10,52 @@ contributors:
...
+restricted to one framework.
 
 _Note that Stencil does **NOT** use Puppeteer for SSR or prerendering._
 
 ## How to Use the Hydrate App
 
-Server side rendering (SSR) can be accomplished in a similar way to prerendering. Instead of using the `--prerender` CLI flag, you can add `{ type: 'dist-hydrate-script' }` to your `stencil.config.js`. This will generate a `hydrate` app in your root project directory that can be imported and used by your Node server.
+Server side rendering (SSR) can be accomplished in a similar way to
+prerendering. Instead of using the `--prerender` CLI flag, you can an output
+target of type `'dist-hydrate-script'` to your `stencil.config.js`, like so:
```
Comment (rwaskiewicz) [inline src/docs/guides/hydrate-app.md:26]: TS?
```suggestion
target of type `'dist-hydrate-script'` to your `stencil.config.ts`, like so:
```

**Review by rwaskiewicz — COMMENTED** [review]
@alicewriteswrongs one small suggestion

File: src/docs/guides/hydrate-app.md (hunk 2 above)
Comment (alicewriteswrongs) [reply src/docs/guides/hydrate-app.md:26]: haha yup

File: src/docs/guides/hydrate-app.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply src/docs/guides/hydrate-app.md:17]: ok good to know! I just tried to re-word things, glad to know they're also accurate 😄

**Pull request merged**